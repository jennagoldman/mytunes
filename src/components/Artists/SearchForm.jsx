import React from 'react';
import PropTypes from 'prop-types';

const SearchForm = ({ searchTerm, onInputChange, onSearch }) => (
  <section>
    <form onSubmit={onSearch}>
      <input value={searchTerm} onChange={onInputChange} />
      <button>Search</button>
    </form>
  </section>
);

SearchForm.propTypes = {
  searchTerm: PropTypes.string,
  onInputChange: PropTypes.func.isRequired,
  onSearch: PropTypes.func
};

export default SearchForm;
