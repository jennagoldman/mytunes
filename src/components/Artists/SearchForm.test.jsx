import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm.jsx';

describe('SearchForm', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<SearchForm searchQuery="electric light orchestra" onInputChange ={() => {}} onSearch={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
