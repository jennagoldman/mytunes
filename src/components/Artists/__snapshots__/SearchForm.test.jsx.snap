// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches a snapshot 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
    />
    <button>
      Search
    </button>
  </form>
</section>
`;
